src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -59,6 +61,15 @@
 LOCK_EXP_BASE = 2.0
 
 
+# Use the lru_cache to keep our per-monitor-locks around in a thread safe
+# manor. We'll use this to ensure that we process checkins for a individual
+# monitor_environment in sequeunce, even though we're multi-threading our
+# consumer.
+@lru_cache(maxsize=100_000, ttl=60)
+def get_monitor_lock(kry: str):
+    return Lock()
```

Comment:
Not sure how safe this actually is. But seems like a decent way to hold a mapping of `key → lock` with a TTL expiration.

Though I think once we un-lock we would ideally remove the lock, so maybe this isn't the perfect solution

Reply:
yeah don't we need to `acquire` the lock as well?